CSharpFunctionalExtensions/Result/Extensions/With/AsyncResultOfT.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace CSharpFunctionalExtensions
+{
+    public static partial class ResultExtensions
+    {
+        public static Task<Result<TResult>> WithMap<T1, T2, TResult>(this Result<T1> a,
```

Comment:
I feel it can be expressed via `SelectMany`

Reply:
If you're able to give an example, I'll be good with that :D